.github/workflows/ansible-test-integration.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: Integration
+on:
+  push:
+    paths:
+    - 'plugins/**'
+    - 'tests/**'
+    - '.github/workflows/ansible-test-integration.yml'
+  pull_request:
+    paths:
+    - 'plugins/**'
+    - 'tests/**'
+    - '.github/workflows/ansible-test-integration.yml'
+
+jobs:
+  pg-14:
+    uses: data-bene/ansible-collection-gh-workflow/.github/workflows/workflow.yml@release/v1
+    with:
+      docker-image: '[ "ubuntu2004" ]'
+      docker-include: '[]'
+      fail-fast: false
+      pre-test-cmd: "sed -i 's/^pg_version:.*/pg_version: \"14\"/g' ./tests/integration/targets/setup_postgresql_db/defaults/main.yml"
+      test-python: false
+      testing-type: integration
+  pg-13:
+    uses: data-bene/ansible-collection-gh-workflow/.github/workflows/workflow.yml@release/v1
+    with:
+      docker-image: '[ "ubuntu2004" ]'
```

Comment:
Are only strings allowed, or does GHA also accept other type of inputs for workflows? (Having to write everything as a string is not really user-friendly...)

Reply:
GHA is a shame in this area. They basically reinvented a parser on top of yaml and limiting features. 

boolean, number or string. Thus JSON, because GHA conveniently (ahem) provides fromJSON(), toJSON() and it allows to work with strings....

https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onworkflow_callinputsinput_idtype

Reply:
Wow, that's really limiting! :(

Reply:
> > ```
> > * license is here because I didn't know where to add it. GPL 3 suggests to show on terminal or an "about" box the copyright, I added it here because it is related to the workflow (action themselves output the © in their output). Its just for being compliant but it is maybe too much ...
> > ```
> 
> I wonder whether it really needs to be printed. Most CLI programs licensed under GPL also only print it when you explicitly ask them (calling with `--version`), and sometimes even don't mention it in that case.
> 
> > > ```
> > > 3. It is probably a good idea (both from environmental impact and RedHat's CI budget) to limit the number of combinations, so that not every combination of ansible-core version, Python version and postgresql version is combined.
> > > ```
> > 
> > 
> > I am not enough involved in ansible requirements to address that. I tend to agree, at least for the environment impact (but remember that the current run is the largest one possible, and often just a subset of the ansible-test will be required). On the number of jobs, it's easy to change. Either in the workflow when calling the action, or (I prefer) in the action having a flag for `full-matrix` or `eco-matrix` (the most important and interesting versions with the smallest matrix). I found disturbing the very large number of versions to support, even old dead and unmaintained like python 2.something, the action to get the versions was here to at least give developers a reference of the largest matrix they are supposed to handle in the plugins. but getting the minimal one is of interest too !
> 
> I'm not sure how this can be properly automated. Picking which combinations are useful is something the collection maintainers are most qualified for, and this is IMO one of the reasons why constructing the matrix should be part of the collection's workflow.

Maybe, from end-user it looks like it's just more fuzzy. So I need to go check that in fact the collections I need really support the same combos, which might be true, or false, until tested. I would much prefer a "stable" vision of collection testing than having to inspect each one.

Based on https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_data/completion/docker.txt maybe it's possible to test only those target python versions that the main distro support by default. (you note that I am not limiting to integration, but also the sanity matrix.)

> > About PostgreSQL version, I fear it is of interest to keep all stable versions in the matrix.
> 
> That's totally fine. It's just not necessary to test every PostgreSQL version with every other ansible-core / Python combination. For older PostgreSQL versions, it suffices if they are tested with exactly one ansible-core / Python version combination, or maybe two (maybe one Python 2 and one Python 3, each with a different ansible-core version?).

Go it, will update in this direction.